recon/src/libp2p/tests.rs

```diff
@@ -26,43 +26,40 @@ use libp2p::swarm::{keep_alive, Swarm, SwarmEvent};
 use libp2p_swarm_test::SwarmExt;
 use quickcheck::QuickCheck;
 use rand::{thread_rng, Rng};
-use std::{
-    num::NonZeroU8,
-    sync::{Arc, Mutex},
-    time::Duration,
-};
+use std::{num::NonZeroU8, time::Duration};
+use tokio::runtime::Runtime;
 use tracing::debug;
 use tracing_test::traced_test;
 
 use crate::{
     libp2p::{stream_set::StreamSet, Behaviour, Config, Event, PeerEvent, PeerStatus},
     recon::{FullInterests, ReconInterestProvider},
-    BTreeStore, Recon, Sha256a,
+    BTreeStore, Client, Recon, Server, Sha256a,
 };
 
 type InterestStore = BTreeStore<Interest, Sha256a>;
 type InterestInterest = FullInterests<Interest>;
 type ReconInterest = Recon<Interest, Sha256a, InterestStore, InterestInterest>;
-type ArcReconInterest = Arc<Mutex<ReconInterest>>;
+type ReconInterestClient = Client<Interest, Sha256a>;
 
-type ModelInterest = ReconInterestProvider<Sha256a, InterestStore, InterestInterest>;
 type ModelStore = BTreeStore<EventId, Sha256a>;
+type ModelInterest = ReconInterestProvider;
 type ReconModel = Recon<EventId, Sha256a, ModelStore, ModelInterest>;
-type ArcReconModel = Arc<Mutex<ReconModel>>;
+type ReconModelClient = Client<EventId, Sha256a>;
 
-type SwarmTest = Swarm<Behaviour<ArcReconInterest, ArcReconModel>>;
+type SwarmTest = Swarm<Behaviour<ReconInterestClient, ReconModelClient>>;
 
 fn random_cid() -> Cid {
-    let mut data = [0u8; 64];
+    let mut data = [0u8; 8];
     thread_rng().fill(&mut data);
     let hash = Code::Sha2_256.digest(&data);
-    Cid::new_v1(0x12, hash)
+    Cid::new_v1(0x00, hash)
 }
 
-fn build_swarm(name: &str, config: Config) -> SwarmTest {
+fn build_swarm(runtime: &Runtime, name: &str, config: Config) -> SwarmTest {
```

Comment:
Why are we using runtime, and not just tokio::spawn?

Reply:
Added a comment in the code.